### PR TITLE
doc for Spaces

### PR DESCRIPTION
**Pull request by julien-c**
Screen capture:

screencapture-localhost-5564-docs-hub-spaces-2021-07-14-14_24_34

### CODE REVIEW

**Comment by cbensimon** [comment]
Really neat !

(I just need to actually implement pre-installed dependencies and finish Secrets management for the doc to be totally accurate)

File: docs/hub/endpoints.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ title: Hugging Face Hub Endpoints
 # Hugging Face Hub API Endpoints
 
 
-We have open endpoints that you can use to retrieve information from the Hub as well as perform certain actions such as creating model repos. We offer a wrapper Python library, [`huggingface_hub`](https://github.com/huggingface/huggingface_hub), that allows easy access to these endpoints. We also provide a webhook to receive real-time incremental info about models. Enjoy!
+We have open endpoints that you can use to retrieve information from the Hub as well as perform certain actions such as creating model repos. We offer a wrapper Python library, [`huggingface_hub`](https://github.com/huggingface/huggingface_hub), that allows easy access to these endpoints. We also provide a [webhook](#webhook) to receive real-time incremental info about models. Enjoy!
```
Comment (cbensimon) [inline docs/hub/endpoints.md:8]: Unrelated ?

File: docs/hub/spaces.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+## What are Spaces?
+
+Spaces are a simple way to host a ML demo app, directly on your user profile or your organization’s hf.co profile.
+
+We support two awesome SDKs that let you build cool apps in Python: **[Streamlit](https://streamlit.io/)** and **[Gradio](https://gradio.app/)**.
+
+**To get started**, simply click on [New Space](https://huggingface.co/new-space) in the top navigation menu, create a new repo of type `Space`, and pick your SDK:
+
+![/docs/assets/hub/new-space.gif](/docs/assets/hub/new-space.gif)
```
Comment (cbensimon) [inline docs/hub/spaces.md:17]: Is it in `gif` format because you'd like an animation / screencast here ?

**Review by cbensimon — APPROVED** [review]
Really neat !

(I just need to actually implement pre-installed dependencies and finish Secrets management for the doc to be totally accurate)

File: docs/hub/spaces.md (hunk 2 above)
Comment (julien-c) [reply docs/hub/spaces.md:17]: yes it already is an animated gif

File: docs/hub/endpoints.md (hunk 1 above)
Comment (julien-c) [reply docs/hub/endpoints.md:8]: yep sorry

**Comment by patrickvonplaten** [comment]
This is awesome - it'd be great to let participants try that out :-) 

In the docs it says that `transfomers` is not pre-installed. Should we have transformers pre-installed as well next to `datasets`?

File: docs/hub/spaces.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+Under the hood, we are storing your code inside a hf.co-hosted git repository, similar to what we're doing for models and datasets. So the same tools you're already used to (git and git-lfs) will also work for Spaces.
```
Comment (patrickvonplaten) [inline docs/hub/spaces.md:19]: (nit)

```suggestion
Under the hood, we are storing your code inside a hf.co-hosted git repository, similar to what we're doing for models and datasets. So the same tools you're already used to (`git` and `git-lfs`) will also work for Spaces.
```

**Review by patrickvonplaten — APPROVED** [review]

**Comment by julien-c** [comment]
> In the docs it says that transfomers is not pre-installed. Should we have transformers pre-installed as well next to datasets?

Our thinking with @cbensimon was to be model-agnostic (the hub is not just for `transformers`), but i agree it'd be convenient to have it by default – maybe we can have it but not mention it in the doc?

File: docs/hub/spaces.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
+---
+title: Documentation for Spaces
+---
+
+<h1>How to get started with Spaces</h1>
+
+> ⚠️ This feature is currently in private beta, reach out to [julien-c](https://huggingface.co/julien-c) if you'd like to try it out ⚠️
```
Comment (osanseviero) [inline docs/hub/spaces.md:7]: I don't think https://huggingface.co/julien-c provides any email to contact you, should this be your email instead?

**Review by osanseviero — APPROVED** [review]
This is awesome! 🎉 

Looking forward to this

**Review by LysandreJik — APPROVED** [review]
Looks great!

File: docs/hub/spaces.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+Streamlit's documentation is at https://docs.streamlit.io/, and Gradio's doc is https://gradio.app/docs.
+
+In the default environment, we're currently running version `"0.79.0"` of Streamlit and version `"2.0.9"` of Gradio.
+
+Our 2 cents:
+- **Gradio** is great if you want to build a super-easy-to-use interface to run a model from just the list of its inputs and its outputs. The Gradio team wrote a great [tutorial on our blog](https://huggingface.co/blog/gradio).
+- **Streamlit** gives you more freedom to build a full-featured Web app from Python, in a _reactive_ way (meaning that code gets re-run when the state of the app changes).
+
+You can also take a look at some sample apps on the [Spaces directory](https://huggingface.co/spaces) to make up your mind.
```
[the remark this answers is not in the document]
Comment (julien-c) [reply docs/hub/spaces.md:37]: it will when this is live (upcoming PR) ;)

File: docs/hub/spaces.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+## What are the RAM and CPU or GPU limitations?
+
+Each environment is currently limited to 16GB RAM and 8 CPU cores.
+
+Some Spaces can have one T4 GPU on a case-by-case basis, contact us if you need one.
+
+## How does it work?
+
+We deploy a containerized version of your code on our Infra, each time you commit. As a sidenote, we have many cool infra challenges to solve, if you'd like to help us, please consider reaching out 🙂.
```
Comment (n1t0) [inline docs/hub/spaces.md:68]: :+1:

**Review by n1t0 — APPROVED** [review]
Awesome, looks great!

File: docs/hub/spaces.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+Our 2 cents:
+- **Gradio** is great if you want to build a super-easy-to-use interface to run a model from just the list of its inputs and its outputs. The Gradio team wrote a great [tutorial on our blog](https://huggingface.co/blog/gradio).
+- **Streamlit** gives you more freedom to build a full-featured Web app from Python, in a _reactive_ way (meaning that code gets re-run when the state of the app changes).
+
+You can also take a look at some sample apps on the [Spaces directory](https://huggingface.co/spaces) to make up your mind.
+
+<!-- Add screencap of listing directory -->
+
+Finally, we've been thinking of providing a way to run **custom apps**, for instance Python server code for the backend + a unified set of widgets/frontend JS code, or even custom Docker image serving. Do get in touch if you would like to build something more custom.
+
```
Comment (jeffboudier) [inline docs/hub/spaces.md:42]:
```suggestion
## Can I create private Spaces?

Yes! Just like with models and datasets, a Hugging Face subscription will allow you to set a new Space as private when creating it. You can create a private Space for your account, or share a private Space within your organization. [Check our our plans](https://huggingface.co/pricing) to get started. 
```

**Comment by jeffboudier** [comment]
Do we, or can we enable Private Spaces for subscribers only? Assuming it's
the case, left a suggestion to message it in the doc.

File: docs/hub/spaces.md (hunk 7 above)
Comment (julien-c) [reply docs/hub/spaces.md:42]: technically aren't restricted to Paid users, but we can still have this wording to promote Pro accounts (it's remaining in other parts of the doc too)

File: docs/hub/spaces.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+## Should I use Streamlit or Gradio?
+
+We recommend you try both as they're both really awesome! 😎
+
+Streamlit's documentation is at https://docs.streamlit.io/, and Gradio's doc is https://gradio.app/docs.
+
+In the default environment, we're currently running version `"0.79.0"` of Streamlit and version `"2.0.9"` of Gradio.
+
+Our 2 cents:
+- **Gradio** is great if you want to build a super-easy-to-use interface to run a model from just the list of its inputs and its outputs. The Gradio team wrote a great [tutorial on our blog](https://huggingface.co/blog/gradio).
```
Comment (julien-c) [inline docs/hub/spaces.md:34]:
```suggestion
- **Gradio** is great if you want to build a super-easy-to-use interface to run a model from just the list of its inputs and its outputs. The Gradio team wrote a great [tutorial on our blog about building GUIs for Hugging Face models](https://huggingface.co/blog/gradio).
```

File: docs/hub/spaces.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+Streamlit's documentation is at https://docs.streamlit.io/, and Gradio's doc is https://gradio.app/docs.
+
+In the default environment, we're currently running version `"0.79.0"` of Streamlit and version `"2.0.9"` of Gradio.
+
+Our 2 cents:
+- **Gradio** is great if you want to build a super-easy-to-use interface to run a model from just the list of its inputs and its outputs. The Gradio team wrote a great [tutorial on our blog about building GUIs for Hugging Face models](https://huggingface.co/blog/gradio).
+- **Streamlit** gives you more freedom to build a full-featured Web app from Python, in a _reactive_ way (meaning that code gets re-run when the state of the app changes).
+
+You can also take a look at some sample apps on the [Spaces directory](https://huggingface.co/spaces) to make up your mind.
+
```
Comment (julien-c) [inline docs/hub/spaces.md:38]:
```suggestion

You can also take a look at some sample apps on the [Spaces directory](https://huggingface.co/spaces) (⚠️ Note: not live yet!) to make up your mind.

```

**Comment by julien-c** [comment]
Thanks everyone for the reviews, I _think_ I addressed everything, but do let me know if i missed anything
